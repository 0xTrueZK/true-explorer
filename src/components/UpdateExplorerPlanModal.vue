<template>
    <v-dialog v-model="dialog" max-width="1800" :persistent="true">
        <v-card>
            <v-card-title>
                Select A Plan
                <v-spacer></v-spacer>
                <v-btn icon @click="close()" ><v-icon>mdi-close</v-icon></v-btn>
            </v-card-title>
            <v-card-text>
                <ul style="list-style: none;" v-if="!user.cryptoPaymentEnabled || user.canTrial" class="mb-4 pl-0">
                    <li v-if="!user.cryptoPaymentEnabled">To setup crypto payment (Explorer 150 or above), reach out to contact@tryethernal.com.</li>
                    <li v-if="user.canTrial">Each plan includes a 7 day free trial - No credit card needed.</li>
                </ul>
                <Explorer-Plan-Selector
                    :pendingCancelation="options.pendingCancelation"
                    :currentPlanSlug="options.currentPlanSlug"
                    :isTrialing="options.isTrialing"
                    :explorerId="options.explorerId"
                    :stripeSuccessUrl="`http://app.${mainDomain}/explorers/${options.explorerId}?justCreated=true`"
                    :stripeCancelUrl="`http://app.${mainDomain}/explorers/${options.explorerId}`"
                    @planUpdated="planUpdated"
                    @planCanceled="planCanceled"
                    @planCreated="planUpdated"></Explorer-Plan-Selector>
            </v-card-text>
        </v-card>
    </v-dialog>
</template>
<script>
import { mapGetters } from 'vuex';
import ExplorerPlanSelector from './ExplorerPlanSelector.vue';

export default {
    name: 'UpdateExplorerPlanModal',
    components: {
        ExplorerPlanSelector
    },
    data: () => ({
        dialog: false,
        resolve: null,
        reject: null,
        options: {},
        refresh: false
    }),
    methods: {
        open(options) {
            this.dialog = true;
            this.options = options;
            return new Promise((resolve, reject) => {
                this.resolve = resolve;
                this.reject = reject;
            });
        },
        planCreated(slug) {
            this.options.currentPlanSlug = slug;
            this.refresh = true;
        },
        planUpdated(slug) {
            this.options.currentPlanSlug = slug;
            this.options.pendingCancelation = false;
            this.refresh = true;
        },
        planCanceled() {
            this.options.pendingCancelation = true;
            this.refresh = true;
        },
        close() {
            this.resolve(this.refresh);
            this.reset();
        },
        reset: function() {
            this.dialog = false;
            this.resolve = null;
            this.reject = null;
            this.refresh = false;
        },
    },
    computed: {
        ...mapGetters([
            'user',
            'mainDomain'
        ])
    }
}
</script>
