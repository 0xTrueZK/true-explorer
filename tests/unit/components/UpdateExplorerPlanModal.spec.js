import MockHelper from '../MockHelper';

import UpdateExplorerPlanModal from '@/components/UpdateExplorerPlanModal.vue';

let helper;
beforeEach(() => {
    jest.clearAllMocks()
    helper = new MockHelper();
});

describe('UpdateExplorerPlanModal.vue', () => {
    it('Should display plan selector', () => {
        const wrapper = helper.mountFn(UpdateExplorerPlanModal, {
            stubs: ['Explorer-Plan-Selector'],
            getters: {
                user: jest.fn().mockReturnValue({ cryptoPaymentEnabled: false, canTrial: true })
            },
            data() {
                return {
                    dialog: true,
                    options: {
                        currentPlanSlug: '100',
                        pendingCancelation: false
                    }
                }
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });
});
