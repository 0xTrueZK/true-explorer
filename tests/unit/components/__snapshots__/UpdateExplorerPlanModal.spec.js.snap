// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateExplorerPlanModal.vue Should display plan selector 1`] = `
<div class="v-dialog__container">
  <div role="dialog" aria-modal="true" class="v-dialog__content v-dialog__content--active" style="z-index: 202;">
    <transition-stub name="dialog-transition" appear="true">
      <div tabindex="0" class="v-dialog v-dialog--active v-dialog--persistent" style="transform-origin: center center; max-width: 1800px;">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title">
            Select A Plan
            <div class="spacer"></div> <button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-close theme--light"></i></span></button>
          </div>
          <div class="v-card__text">
            <ul class="mb-4 pl-0" style="list-style: none;">
              <li>To setup crypto payment (Explorer 150 or above), reach out to contact@tryethernal.com.</li>
              <li>Each plan includes a 7 day free trial - No credit card needed.</li>
            </ul>
            <explorer-plan-selector-stub bestvalueslug="explorer-150" currentplanslug="100" stripesuccessurl="http://app.tryethernal.com/explorers/undefined?justCreated=true" stripecancelurl="http://app.tryethernal.com/explorers/undefined"></explorer-plan-selector-stub>
          </div>
        </div>
      </div>
    </transition-stub>
  </div>
</div>
`;
